File: trip-kit-booking/src/main/java/com/skedgo/android/tripkit/booking/BookingAction.java
package com.skedgo.android.tripkit.booking;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

public class BookingAction implements Parcelable {
  public static final Creator<BookingAction> CREATOR = new Creator<BookingAction>() {
    @Override
    public BookingAction createFromParcel(Parcel in) {
      return new BookingAction(in);
    }

    @Override
    public BookingAction[] newArray(int size) {
      return new BookingAction[size];
    }
  };
  @SerializedName("title")
  private String title;
  @SerializedName("enabled")
  private boolean enable = true;
  @SerializedName("url")
  private String url;
  @SerializedName("hudText")
  private String hudText;

  public BookingAction() {

  }

  private BookingAction(Parcel in) {
    title = in.readString();
    enable = Boolean.valueOf(in.readString());
    url = in.readString();
    hudText = in.readString();
  }

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public boolean isEnable() {
    return enable;
  }

  public void setEnable(boolean enable) {
    this.enable = enable;
  }

  public String getUrl() {
    return url;
  }

  public void setUrl(String url) {
    this.url = url;
  }

  @Nullable
  public String getHudText() {
    return hudText;
  }

  @Override
  public int describeContents() {
    return 0;
  }

  @Override
  public void writeToParcel(Parcel dest, int flags) {
    dest.writeString(title);
    dest.writeString(String.valueOf(enable));
    dest.writeString(url);
    dest.writeString(hudText);
  }
}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/BookingAction.kt
package com.skedgo.tripkit.booking

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.SerializedName

data class BookingAction(
    @SerializedName("title") var title: String? = null,
    @SerializedName("enabled") var enable: Boolean = true,
    @SerializedName("url") var url: String? = null,
    @SerializedName("hudText") var hudText: String? = null,
    @SerializedName("done") var done: Boolean = false
) : Parcelable {

    private constructor(parcel: Parcel) : this() {
        title = parcel.readString()
        enable = parcel.readString()?.toBoolean() ?: true
        url = parcel.readString()
        hudText = parcel.readString()
        done = parcel.readString()?.toBoolean() ?: false
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
        parcel.writeString(enable.toString())
        parcel.writeString(url)
        parcel.writeString(hudText)
        parcel.writeString(done.toString())
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<BookingAction> {
        override fun createFromParcel(parcel: Parcel): BookingAction {
            return BookingAction(parcel)
        }

        override fun newArray(size: Int): Array<BookingAction?> {
            return arrayOfNulls(size)
        }
    }
}
